src/posix/platform/platform-posix.h

```diff
@@ -102,6 +107,84 @@ enum
+                otLogCritPlat("exit(%d): %s: %s", aExitCode, __func__, strerror(errno));       \
+            }                                                                                  \
+            exit(aExitCode);                                                                   \
+        }                                                                                      \
+    } while (false)
+
+/**
+ * This macro checks for the specified error code, which is expected to commonly be successful,
+ * and both records exit status and terminates the program if the error code is unsuccessful.
+ *
+ * @param[in]  aError  An error code to be evaluated against OT_ERROR_NONE.
+ *
+ */
+#define SuccessOrDie(aError)                                                                            \
+    do                                                                                                  \
+    {                                                                                                   \
+        if (aError != OT_ERROR_NONE)                                                                    \
+        {                                                                                               \
+            uint8_t exitCode;                                                                           \
+            exitCode = (aError == OT_ERROR_INVALID_ARGS) ? OT_EXIT_INVALID_ARGUMENTS : OT_EXIT_FAILURE; \
+            fprintf(stderr, "exit(%d): %s: %s\r\n", exitCode, __func__, otThreadErrorToString(aError)); \
+            otLogCritPlat("exit(%d): %s: %s", exitCode, __func__, otThreadErrorToString(aError));       \
+            exit(exitCode);                                                                             \
+        }                                                                                               \
+    } while (false)
+
+/**
+ * This macro unconditionally both records exit status and terminates the program.
+ *
+ * @param[in]   aExitCode   The exit code.
+ *
+ */
+#define DieNow(aExitCode) VerifyOrDie(false, aExitCode)
+
+/**
+ * This macro unconditionally both records exit status and exit message and terminates the program.
+ *
+ * @param[in]   aMessage    The exit message.
+ * @param[in]   aExitCode   The exit code.
+ *
+ */
+#define DieNowWithMessage(aMessage, aExitCode)                                                           \
+    do                                                                                                   \
+    {                                                                                                    \
+        if (errno == 0)                                                                                  \
+        {                                                                                                \
+            fprintf(stderr, "exit(%d): %s: %s\r\n", aExitCode, __func__, aMessage);                      \
```

Comment:
How about we introduce a command line option `-v` so set `LOG_PERROR` when initializing syslog?

Reply:
Current source code of posix-app doesn't have a  uniform way to control how to display what kind of  information on the console. I think it is better to submit another PR to process this issue.

Reply:
Agree.